server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -588,7 +588,8 @@ private boolean isCheckOriginEnabled(String zoneId) {
 
     private void checkOrigin(String origin, String zoneId) {
         Integer count = jdbcTemplate.queryForObject("select count(*) from identity_provider where origin_key=? and identity_zone_id=? ", Integer.class, origin, zoneId);
-        if (count == 0) {
+        int c = (count == null) ? 0 : count;
```

Comment:
It seems that the query string is existing code. This query operation can be optimized by using `exists` instead of `count` though, because we just need to know whether there is at least 1 such record that meets the condition or not. For example, in mysql, the query can be as follows instead:

`select exists (select * from identity_provider where origin_key=? and identity_zone_id=?)`

And then the result can be assigned to a `boolean` variable, removing the need to check for null Integer value.

Reply:
In this case, count can only be 0 or 1, and there is a matching index on table identity_provider.
https://github.com/cloudfoundry/uaa/blob/develop/server/src/main/resources/org/cloudfoundry/identity/uaa/db/postgresql/V2_0_0__Multitenancy.sql#L37
So I don't think that the query can be optimized with respect to runtime.

Reply:
@klaus-sap , do you not agree that exists query is more efficient than count query? Even if the count can be only 1 or 0, count query needs to go through all records while exists query does not, right?

Reply:
I see. There is unique index on `identity_provider (identity_zone_id,origin_key)`. So in theory, not much to gain on the SQL side then. In that case, the only benefit of changing the query is likely not having to covert from Integer to int, and I am OK if you don't think that is important enough to make further change.